src/runtime/virtcontainers/utils/utils.go

```diff
@@ -397,26 +397,38 @@ func MkdirAllWithInheritedOwner(path string, perm os.FileMode) error {
 	paths := getAllParentPaths(path)
 	for _, curPath := range paths {
 		info, err := os.Stat(curPath)
-
-		if err != nil {
-			if err = os.Mkdir(curPath, perm); err != nil {
-				return fmt.Errorf("mkdir call failed: %v", err.Error())
+		if err == nil {
+			if !info.IsDir() {
+				return &os.PathError{Op: "mkdir", Path: curPath, Err: syscall.ENOTDIR}
+			}
+			if stat, ok := info.Sys().(*syscall.Stat_t); ok {
+				uid = int(stat.Uid)
+				gid = int(stat.Gid)
+				continue
+			} else {
+				return fmt.Errorf("fail to retrieve the uid and gid of path %s", curPath)
 			}
+		}
+
+		err = os.Mkdir(curPath, perm)
+		if err == nil {
 			if err = syscall.Chown(curPath, uid, gid); err != nil {
 				return fmt.Errorf("chown syscall failed: %v", err.Error())
 			}
 			continue
 		}
 
+		if !os.IsExist(err) {
```

Comment:
You dont need to check for if the directory exists here again. Just check for in the else clause of the previous if statement.

Reply:
Thanks @amshinde, updated.